src/compiler/visitor.ts

```diff
@@ -268,7 +268,7 @@ namespace ts {
                     nodesVisitor((<PropertyDeclaration>node).decorators, visitor, isDecorator),
                     nodesVisitor((<PropertyDeclaration>node).modifiers, visitor, isModifier),
                     visitNode((<PropertyDeclaration>node).name, visitor, isPropertyName),
-                    visitNode((<PropertyDeclaration>node).questionToken, tokenVisitor, isToken),
+                    visitNode((<PropertyDeclaration>node).questionToken, tokenVisitor, isToken) || visitNode((<PropertyDeclaration>node).exclamationToken, tokenVisitor, isToken),

```

Comment:
Use `,` instead of `||`

Reply:
Actually, `||` would have been better in this case because the argument only allows one or the other. I've suggested a more concise approach in another comment.